digitalearthau/paths.py

```diff
@@ -299,3 +318,37 @@ def trash_uri(uri: str, dry_run=False, log=_LOG):
         if not trash_path.parent.exists():
             os.makedirs(str(trash_path.parent))
         os.rename(str(base_path), str(trash_path))
+
+
+def get_product_work_directory(
+        output_product: str,
+        time=datetime.datetime.utcnow(),
```

Comment:
At the moment I'd only expect this function to be called once.

But it's probably a gotcha in the future if anyone tried to call it a few times from the some process and they were given identical timestamps.

Reply:
Good catch. It'll fail in subsequent calls because the directory already exists, but that's not an ideal failure case. It'll also fail for the same reason if called multiple times within the same second for the same type of product/process (due to directory name granularity of HHMMSS). 

It's good enough for our current cli-usage of creating infrequent large jobs (and failures will be loud, not silent), so I don't think it's a blocker, but will need some improvement for wider use cases.